External/FEXCore/Source/Interface/Core/ArchHelpers/Arm64Emitter.cpp

```diff
@@ -268,14 +273,30 @@ void Arm64Emitter::LoadConstant(ARMEmitter::Size s, ARMEmitter::Register Reg, ui
     }
   }
   else {
-    movz(s, Reg, (Constant) & 0xFFFF, 0);
-    for (int i = 1; i < Segments; ++i) {
-      uint16_t Part = (Constant >> (i * 16)) & 0xFFFF;
+    int CurrentSegment = 0;
+    for (; CurrentSegment < Segments; ++CurrentSegment) {
+      uint16_t Part = (Constant >> (CurrentSegment * 16)) & 0xFFFF;
+      if (Part) {
+        movz(s, Reg, Part, CurrentSegment * 16);
+        ++CurrentSegment;
+        ++NumMoves;
+        break;
+      }
+    }
+
+    for (; CurrentSegment < Segments; ++CurrentSegment) {
+      uint16_t Part = (Constant >> (CurrentSegment * 16)) & 0xFFFF;
       if (Part) {
-        movk(s, Reg, Part, i * 16);
+        movk(s, Reg, Part, CurrentSegment * 16);
         ++NumMoves;
       }
     }
+
+    if (NumMoves == 0) {
+      // If we didn't move anything that means this is a zero move. Special case this.
+      mov(s, Reg, ARMEmitter::Reg::zr);
```

Comment:
Unfortunately `mov reg, zr` is slow on Apple CPUs, but as far as I know, `mov reg, #0x0` should be fast everywhere?

Reply:
Good idea. Switched over to move of constant zero.